catalyst_voices/apps/voices/lib/pages/campaign/admin_tools/campaign_admin_tools_events.dart

```diff
@@ -136,19 +266,77 @@ class _EventTimelapseControls extends StatelessWidget {
               color: Theme.of(context).colors.elevationsOnSurfaceNeutralLv1Grey,
             ),
             alignment: Alignment.center,
-            // TODO(dtscalac): implement timer ticking
-            child: const Text('-5s'),
+            child: TextField(
+              controller: _timerController,
+              onChanged: (_) => _onTransitionDelayChanged(),
+              enabled: _enabled,
+              decoration: null,
+              textAlign: TextAlign.center,
+            ),
           ),
           Expanded(
             child: VoicesTextButton(
               leading: VoicesAssets.icons.fastForward.buildIcon(),
+              onTap: widget.onNextStage,
               child: Text(context.l10n.next),
-              // TODO(dtscalac): implement button action
-              onTap: () {},
             ),
           ),
         ],
       ),
     );
   }
+
+  void _restartRefreshTimer() {
+    _refreshTimer?.cancel();
+    _refreshTimer = Timer.periodic(
+      const Duration(seconds: 1),
+      (_) => _refresh(),
+    );
+  }
+
+  void _refresh() {
+    final now = DateTimeExt.now();
+    final nextStageTransitionAt = widget.nextStageTransitionAt;
+
+    if (nextStageTransitionAt != null && nextStageTransitionAt.isAfter(now)) {
+      final remainingDelay = nextStageTransitionAt.difference(now);
+      _updateTimerController(remainingDelay, enabled: false);
+    } else {
+      _updateTimerController(widget.stageTransitionDelay, enabled: true);
+    }
+  }
+
+  void _updateTimerController(Duration duration, {required bool enabled}) {
+    // only update if the duration is different,
+    // otherwise we might be overwriting local user edits
+    if (_stageTransitionDelay != duration) {
+      final seconds =
+          (duration.inMilliseconds / Duration.millisecondsPerSecond).ceil();
```

Comment:
how bout 

```dart
final seconds = duration.inSeconds;
```

Reply:
`duration.inSeconds` gives you only full seconds, this causes 4.99s to show as 4s while in reality it's much closer to 5s than to 4s

Reply:
Also I didn't want to use round, because for 0.49s it would show as 0s and this looks strange cause the user thinks the timer is already done but why hasn't it fired?

Reply:
okey, makes sense